src/problems.jl

```diff
@@ -224,7 +224,8 @@ function build_solution(
         interp.cache,
         prob.manifold,
     )
-    return ODESolution{T,1}(
```

Comment:
Unless your solutions were scalar valued this was never correct.

Reply:
If `ndims` is required to be defined to use `ODESolution` then we can't use it here.

Reply:
And it looks like if we can't use `ODESolution`, `SciMLBase` has almost nothing that can be used in `ManifoldDiffEq`.

Reply:
I am not sure, whether maybe the solution in this PR is enough, but I also could not find what `N` actually refers to by looking at either https://docs.sciml.ai/DiffEqDocs/stable/types/ode_types/#SciMLBase.ODESolution or (had to look for that manually since it was not linked) https://docs.sciml.ai/SciMLBase/stable/interfaces/Solutions/#SciMLBase.AbstractODESolution or (same manual search) https://docs.sciml.ai/SciMLBase/stable/interfaces/Solutions/#SciMLBase.AbstractTimeseriesSolution ... (and then I got a bit tired).

Reply:
Any solution that uses `ndims` is at best a workaround since `ndims` is not a valid concept for points on a manifold or tangent vectors.

Reply:
I do agree.